plugins/newsdownloader.koplugin/feed_source.lua

```diff
@@ -0,0 +1,403 @@
+        )
+    end
+
+    return initialized_feeds
+end
+
+-- This function contacts the feed website and attempts to get
+-- the RSS/Atom document with a list of the latest items.
+function FeedSource:fetchDocumentByUrl(url)
+    local document
+    -- Get the XML document representing the feed
+    local ok, response = pcall(function()
+            local success, content = NewsHelpers:getUrlContent(url)
+            if (success) then
+                return content
+            else
+                error("Failed to download content for url:", url)
+            end
+    end)
+    -- Check to see if a response is available to deserialize.
+    if ok then
+        -- Deserialize the XML document into something Lua can use
+        document = NewsHelpers:deserializeXMLString(response)
+    end
+    -- Return the document or any errors that may have occured
+    if ok or document then
+        return document
+    else
+        if not ok then
+            error("(Reason: Failed to download feed document)")
+        else
+            error("(Reason: Error during feed document deserialization)")
+        end
+    end
+end
+
+-- Supply this method with the XML document returned by the feed,
+-- and it will initialized the document by extracting the feed title,
+-- feed items, and items count.
+function FeedSource:initializeDocument(document)
+    local feed_title
+    local feed_items
+    local total_items
+
+    local ok = pcall(function()
+            return self:getFeedType(
+                document,
+                function()
+                    -- RSS callback
+                    feed_title = util.htmlEntitiesToUtf8(document.rss.channel.title)
+                    feed_items = document.rss.channel.item
+                    total_items = #document.rss.channel.item
+                end,
+                function()
+                    -- Atom callback
+                    feed_title = FeedSource:getFeedTitle(document.feed.title)
+                    feed_items = document.feed.entry
+                    total_items = #document.feed.entry
+                end
+            )
+    end)
+
+    if ok then
+        document.title = feed_title
+        document.items = feed_items
+        document.total_items = total_items
+        return document
+    else
+        error("Could not initialize feed document")
+    end
+end
+
+function FeedSource:getItemsContent(feed, progress_callback, error_callback)
+    local limit = tonumber(feed.config.limit)
+    local total_items = (limit == 0) and
+        feed.document.total_items or
+        limit
+    local initialized_feed_items = {}
+    -- Download each ite0m in the feed
+    for index, item in pairs(feed.document.items) do
+        -- If limit has been met, stop downloading feed.
+        if limit ~= 0 and index - 1 == limit then
+            break
+        end
+        -- Display feedback to user.
+        progress_callback(T(
+            _("Getting item %1 of %2 from %3"),
```

Comment:
Did any of these strings change? (If they're existing strings I won't pay as much attention.)

Reply:
I do believe the strings *have* change.